user-guide/python.rst

```diff
@@ -270,225 +240,129 @@ Please click on the link indicated to see examples of how to use the `PyTorch an
 More detail on the Cirrus GPU nodes can be found at https://cirrus.readthedocs.io/en/main/user-guide/gpu.html .
 
 
-Custom Miniconda3 Environments
-------------------------------
+Extending a centrally-installed Miniconda3 environment
+------------------------------------------------------
+
+This section shows how to setup a local custom Python environment such that it extends a centrally-installed Miniconda module.
+By extend, we mean being able to install packages locally that are not provided by the Miniconda module. This is needed because
+some packages such as ``mpi4py`` must be built specifically for the Cirrus system and so are best provided centrally.
+
+The first action to take is to decide which ``python`` module to extend, e.g., ``python/3.9.13-gpu`` (you can run
+``module avail python`` to list all the available ``python`` modules).
 
-To setup a custom Python environment including packages that are not in the
-central installation, the simplest method is to install Miniconda locally within
-your project area on Cirrus.
+.. code-block:: bash
 
-Installing Miniconda3
-~~~~~~~~~~~~~~~~~~~~~
+    [auser@cirrus-login1 auser]$ module load python/3.9.13-gpu
 
-First, you should download Miniconda. You can use ``wget`` to do this.
+Loading the python module above will set a number of environment variables such as ``MINICONDA3_PYTHON_VERSION`` and
+``MINICONDA3_PYTHON_LABEL``. This can be confirmed by looking at the output from ``module show python/3.9.13-gpu``.
 
-::
+.. code-block:: bash
 
-   [user@cirrus-login1 ~]$ wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+    /mnt/lustre/indy2lfs/sw/modulefiles/python/3.9.13-gpu:
 
-You can find links to the various miniconda versions on the Miniconda website.
+    conflict	python
+    setenv	MINICONDA3_PYTHON_VERSION 3.9.13
+    setenv      MINICONDA3_PYTHON_LABEL python3.9
+    ...
+    setenv	MINICONDA3_BIN_PATH /mnt/lustre/indy2lfs/sw/miniconda3/4.12.0-py39-gpu/bin
 
-* https://conda.io/miniconda.html
+The *local* packages will be installed using ``pip``. Now, as the ``/home`` file system is not available on the compute nodes,
+you will need to modify the default install location that ``pip`` uses to point to a location on ``/work``. To do this, you set
+the ``PYTHONUSERBASE`` environment variable to point to the location on ``/work`` where you intend to install your local virtual
+Python environment, which we are calling ``myvenv`` for purposes of illustration.
 
-For Cirrus, you should use the Linux 64-bit installer.
+.. code-block:: bash
 
-Once you have downloaded the installer, you can run it via ``bash``.
+    export PYTHONUSERBASE=/work/x01/x01/auser/myvenv
```

Comment:
Shall that be changed by something generic like `${HOME/home/work}/myenv`?
(not 100% sure myself as, while being convenient and portable, it obfuscates a bit the path)

Reply:
I've added a note that explains this.